payments-core/src/test/java/com/stripe/android/paymentsheet/flowcontroller/DefaultFlowControllerTest.kt

```diff
@@ -409,6 +409,31 @@ internal class DefaultFlowControllerTest {
         flowController.confirm()
     }
 
+    @Test
+    fun `confirmPaymentSelection() with generic payment method should start paymentController`() =
+        runBlockingTest {
+            flowController.confirmPaymentSelection(
+                NEW_CARD_PAYMENT_SELECTION,
+                InitData(
+                    PaymentSheetFixtures.CONFIG_CUSTOMER,
+                    PaymentSheetFixtures.PAYMENT_INTENT_CLIENT_SECRET,
+                    PaymentIntentFixtures.PI_REQUIRES_PAYMENT_METHOD,
+                    listOf(PaymentMethod.Type.Card),
+                    PAYMENT_METHODS,
+                    SavedSelection.PaymentMethod(
+                        id = "pm_123456789"
+                    ),
+                    isGooglePayReady = false
+                )
+            )
+
+            verify(paymentController).startConfirmAndAuth(
+                any(),
```

Comment:
`#confirmPaymentSelection` does calculate the parameters passed to `startConfirmAndAuth`, can we assert the actual values here instead of using wildcard?